### DISCUSSION TITLE
Ignore _mock files during commit?

### DISCUSSION THREAD

**Discussion by ghost** [Dynatrace/dynatrace-configuration-as-code#196] [Archive] [answered]
Should we commit the `_mock` files or not? I believe they're autogenerated and thus can be added to `.gitignore`?

**Comment by tatomir146** [comment, marked as answer]
You actually have to generate mock files.
```
### Test Mocks

Go Mockgen is used for some generated mock files. They are not generated every time, but rather checked in, so be careful
when introducing changes to objects that are mocked. You will have to regenerate and probably manually modify them to remove
e.g. the reference to the module.
​To explicitly generate the mocked files, run `make mocks` in the root folder.
```

@didiladi adding this as a build step would be possible, but what in case there is a error during `go generate`?

**Reply by patrickpichler** [reply]
Actually this is already done in a build step. 

The `test` goal in the Makefile depends on the `mocks`
goal, which run generate on the whole project. 

This means, I guess we can ignore mock files, as they
are regenerated anyway during each build.

**Reply by tatomir146** [reply]
@patrickpichler can you please create a ticket and archive this discussion? Then it would also partially solve #191.

**Reply by patrickpichler** [reply]
Ticket to track progress #220

How can we actually archive discussions?
Simply lock it?

**Reply by patrickpichler** [reply]
Nevermind I found out how ^^
